src/Nox.Types/Types/Color/Color.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System.Linq;
+
+namespace Nox.Types;
+
+/// <summary>
+/// Represents a Nox <see cref="Color"/> type and value object. 
+/// </summary>
+public class Color : ValueObject<(byte Alpha, byte Red, byte Green, byte Blue), Color>
+{
+    public Color() { Value = (Alpha = 0, Red = 0, Green = 0, Blue = 0); }
+
+    public byte Alpha
+    {
+        get => Value.Alpha;
+        private set => Value = (Alpha: value, Red: Value.Red, Green: Value.Green, Blue: Value.Blue);
+    }
+
+    public byte Red
+    {
+        get => Value.Red;
+        private set => Value = (Alpha: Value.Alpha, Red: value, Green: Value.Green, Blue: Value.Blue);
+    }
+
+    public byte Green
+    {
+        get => Value.Green;
+        private set => Value = (Alpha: Value.Alpha, Red: Value.Red, Green: value, Blue: Value.Blue);
+    }
+
+    public byte Blue
+    {
+        get => Value.Blue;
+        private set => Value = (Alpha: Value.Alpha, Red: Value.Red, Green: Value.Green, Blue: value);
+    }
+
+    /// <summary>
+    /// Validates a <see cref="Color"/> object.
+    /// </summary>
+    /// <returns>true if the <see cref="Color"/> value is valid.</returns>
+    internal override ValidationResult Validate()
+    {
+        var result = base.Validate();
+
+        if (System.Drawing.Color.FromArgb(Alpha, Red, Green, Blue).IsEmpty)
```

Comment:
Checking in the [Color.cs](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Drawing.Primitives/src/System/Drawing/Color.cs) code I couldn't see any way of `IsEmpty` ever being true, so it doesn't validate anything for us.
But `FromArgb` can throw System.ArgumentException in some situations, but since this is a validate method that adds the validations to out Error array, we should prevent it from throwing an exception.
I think we could do something like `try { System.Drawing.Color.FromArgb(Alpha, Red, Green, Blue) } catch ArgumentException e { result.Errors.Add(new ValidationFailure(e.Message); }` .